hooks/hookexecution/outcome.go

```diff
@@ -77,14 +78,15 @@ type GroupOutcome struct {
 type HookOutcome struct {
 	// ExecutionTime is the execution time of a specific hook without applying its result.
 	ExecutionTime
-	AnalyticsTags hookanalytics.Analytics `json:"analytics_tags"`
-	HookID        HookID                  `json:"hook_id"`
-	Status        Status                  `json:"status"`
-	Action        Action                  `json:"action"`
-	Message       string                  `json:"message"` // arbitrary string value returned from hook execution
-	DebugMessages []string                `json:"debug_messages,omitempty"`
-	Errors        []string                `json:"-"`
-	Warnings      []string                `json:"-"`
+	AnalyticsTags hookanalytics.Analytics    `json:"analytics_tags"`
+	HookID        HookID                     `json:"hook_id"`
+	Status        Status                     `json:"status"`
+	Action        Action                     `json:"action"`
+	Message       string                     `json:"message"` // arbitrary string value returned from hook execution
+	DebugMessages []string                   `json:"debug_messages,omitempty"`
+	Errors        []string                   `json:"-"`
+	Warnings      []string                   `json:"-"`
+	SeatNonBid    openrtb_ext.NonBidsWrapper `json:"-"`
```

Comment:
Why is this not a slice of SeatNonBid objects?

Reply:
We believe, building the standard `openrtb_ext.SeatNonBid` is expensive because each `nonbid` needs to be mapped against respective `seat` and `impid`. 
Similarly, obtaining standard `openrtb_ext.SeatNonBid`  object from each stageOutcome and again building the final `openrtb_ext.SeatNonBid`  would add additional computational overhead. 
Example - 
Assume, RawBidderResponse hook returns
```
SeatNonBid: []openrtb_ext.SeatNonBid{
{
	NonBid: []openrtb_ext.NonBid{
		{
			ImpId:      "imp1",
			StatusCode: 100,
		}
	},
	Seat: "pubmatic",
}
```
& AuctionResponseHook hook returns 
```
SeatNonBid: []openrtb_ext.SeatNonBid{
{
	NonBid: []openrtb_ext.NonBid{
		{
			ImpId:      "imp2",
			StatusCode: 100,
		}
	},
	Seat: "pubmatic",
}
```
But we need to combine them for bidresponseExt.Prebid.SeatNonBid and auctionObject.SeatNonBid like 
```
SeatNonBid: []openrtb_ext.SeatNonBid{
{
	NonBid: []openrtb_ext.NonBid{
                {
			ImpId:      "imp1",
			StatusCode: 100,
		},
		{
			ImpId:      "imp2",
			StatusCode: 100,
		}
	},
	Seat: "pubmatic",
}

```
But, if we allow hookOutcome to return `openrtb_ext.NonBidsWrapper/openrtb_ext.NonBidCollection`   then it become easy to add non-bids by using `SeatNonBid.AddBid()` function.
To combine the SeatNonBids from multiple hookOutcomes, we can use `SeatNonBid.Append()` function. 
Please refer - `getNonBidsFromStageOutcomes` function which does the same.